src/pages/profile-page/profile-page.js

```diff
@@ -0,0 +1,312 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { TextField, Button } from '@material-ui/core';
+import { useStyles } from './profile-page.styles';
+import ProfilePicture from '../../images/profile.png';
+import {
+  updateUser,
+  sendConfirmationEmail,
+  recoverUser
+} from '../../redux/user/user.actions';
+import { Loader } from '../../components/loader/loader';
+import {
+  PROFILE_DATA,
+  PROFILE_PASSWORD_CHANGE,
+  PROFILE_EMAIL_CONFIRM
+} from '../../translations/user.translations';
+
+const ProfilePage = () => {
+  const classes = useStyles();
+
+  const {
+    userData,
+    userLoading,
+    language,
+    confirmationEmailSent,
+    userRecovered
+  } = useSelector(({ User, Language }) => ({
+    userData: User.userData,
+    userLoading: User.userLoading,
+    language: Language.language,
+    confirmationEmailSent: User.confirmationEmailSent,
+    userRecovered: User.userRecovered
+  }));
+
+  const dispatch = useDispatch();
+
+  const [user, setUser] = useState(userData);
+  const [isEdited, setIsEdited] = useState(userData);
+  const [userImageUrl, setUserImageUrl] = useState(null);
+  const [upload, setUpload] = useState(null);
+
+  const handleChange = (e, innerProp) => {
+    const inputName = e.target.name;
+    const inputValue = e.target.value;
+    if (innerProp) {
+      return setUser({
+        ...user,
+        [innerProp]: { ...user[innerProp], [inputName]: inputValue }
+      });
+    }
+    setUser({ ...user, [inputName]: inputValue });
+  };
+
+  const handleImageLoad = (e) => {
+    if (e.target.files && e.target.files[0]) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        setUserImageUrl(e.target.result);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+      setUpload(e.target.files[0]);
+    }
+  };
+
+  const handleConfirmation = () => {
+    dispatch(sendConfirmationEmail({ email: userData.email, language }));
+  };
+
+  const handleSaveUser = () => {
+    delete user.token;
+    dispatch(updateUser({ user, id: user._id, upload }));
+  };
+
+  const handleImageError = (e) => {
+    e.target.src = ProfilePicture;
+  };
+
+  const handlePasswordChange = () => {
+    dispatch(recoverUser({ email: userData.email, language }));
+  };
+
+  useEffect(() => {
+    if (
+      user.address &&
+      Object.keys(user.address).every((key) => !user.address[key])
+    ) {
+      user.address = null;
+    }
+  }, [user.address]);
+
+  useEffect(() => {
+    if (JSON.stringify(user) === JSON.stringify(userData)) {
```

Comment:
It is not the best idea because in an object there is no guarantee of the order of the keys when you stringify it. So, please, iterate per each key and compare values for each field

Reply:
user is a userData's clone, so there is no need for complex comparison logic